### ISSUE TITLE
can not insert to the newly database

### ISSUE THREAD

**Issue by CNLHC** [openGemini/openGemini#160]
### Describe the bug（Bug 描述）

Inserting into the newly created database will cause the server to panic.

```
goroutine 189 [running]:
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/fs.mmapFile({0x106392418?, 0x14000010240?}, 0x54)
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/fs/reader_at.go:125 +0xbc
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/fs.MustOpenReaderAt({0x14000052c60, 0x81})
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/fs/reader_at.go:94 +0x1f4
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.openFilePart({0x140001f0780?, 0x140000503f0?})
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/part.go:82 +0x188
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).mergeParts(0x14000309440, {0x140003860a0?, 0x1, 0x1}, 0x0?, 0x1)
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:967 +0xb3c
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).mergeRawItemsBlocks(0x14000309440, {0x14000386098, 0x1, 0x1}, 0x0)
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:669 +0x300
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*rawItemsShards).flush(0x140003094c0, 0x14000309440, 0x0?)
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:613 +0x120
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).flushRawItems(...)
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:602
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).rawItemsFlusher(0x14000309440)
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:507 +0x74
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).startRawItemsFlusher.func1()
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:494 +0x24
created by github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).startRawItemsFlusher
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:493 +0x70
```

### To Reproduce（Bug 复现步骤）

1. build and run server
```
#ef1999f8271c8c95603db9bd80ab383439732da5
git clone https://github.com/openGemini/openGemini
cd openGemini
python build.py --arch arm64
cd ./build
./ts-server --config ../config/openGemini.singlenode.conf
```

2. create a database and insert a point
```
./ts-cli -host 127.0.0.1 -port 8086 -database bench
>> create database bench;

>> insert sensor,farmID=f1,deviceID=d0 sensorID=20,value=50.98
```

### Expected behavior（期望结果）

The point was inserted into the database.

### Screenshots（屏幕截图）

![image](https://user-images.githubusercontent.com/21005146/229070612-330ad13a-a238-4115-875a-77598b60e0b8.png)

### Logs（完整的错误日志）

```
./ts-server --config ../config/openGemini.singlenode.conf

 _________   ______   ____    ____  ________  _________     _       
|  _   _  |.' ____ \ |_   \  /   _||_   __  ||  _   _  |   / \      
|_/ | | \_|| (___ \_|  |   \/   |    | |_ \_||_/ | | \_|  / _ \     
    | |     _.____.    | |\  /| |    |  _| _     | |     / ___ \
   _| |_   | \____) | _| |_\/_| |_  _| |__/ |   _| |_  _/ /   \ \_
  |_____|   \______.'|_____||_____||________|  |_____||____| |____|

2023-03-31 16:39:46.246059 +0800 CST m=+0.025839834 TSMeta starting

 _________   ______    ______     ___    _____     
|  _   _  |.' ____ \ .' ____ \  .'   '.  |_   _|
|_/ | | \_|| (___ \_|| (___ \_|/  .-.  \   | |
    | |     _.____.  _.____.   | |   | |   | |   _
   _| |_   | \____) || \____) |\   -'  \_ _| |__/ |
  |_____|   \______.' \______.' \__.\__| |________|

SetQueryParallel: 10
2023-03-31 16:39:47.977113 +0800 CST m=+1.759008709 TSSQL starting

 _________   ______    ______   _________    ___   _______     ________  
|  _   _  |.' ____ \ .' ____ \ |  _   _  | .'   '.|_   __ \   |_   __  |
|_/ | | \_|| (___ \_|| (___ \_||_/ | | \_|/  .-.  \ | |__) |    | |_ \_|
    | |     _.____'.  _.____'.     | |    | |   | | |  __ /     |  _| _
   _| |_   | \____) || \____) |   _| |_   \  '-'  /_| |  \ \_  _| |__/ |
  |_____|   \______.' \______.'  |_____|   '.___.'|____| |___||________|


2023-03-31 16:39:48.07822 +0800 CST m=+1.860233251 TSStore starting
2023-03-31T08:39:48.089Z        info    VictoriaMetrics/lib/mergeset/table.go:266       opening table "/tmp/openGemini/data/data/bench/0/autogen/index/1_1679875200000000000_1680480000000000000/mergeset"...
2023-03-31T08:39:48.108Z        info    /Users/liuhancheng/go/pkg/mod/github.com/!victoria!metrics/!victoria!metrics@v1.67.0/lib/memory/memory.go:37    limiting caches to 41231686041 bytes, leaving 27487790695 bytes to the OS according to -memory.allowedPercent=60
panic: runtime error: slice bounds out of range [:84] with capacity 0

goroutine 95 [running]:
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/fs.mmapFile({0x103a96418?, 0x140004ba1b8?}, 0x54)
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/fs/reader_at.go:125 +0xbc
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/fs.MustOpenReaderAt({0x14000342990, 0x81})
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/fs/reader_at.go:94 +0x1f4
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.openFilePart({0x14000148f80?, 0x77?})
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/part.go:82 +0x188
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.openParts({0x14000050460, 0x62}, 0x140004808f0)
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:1102 +0x4b4
github.com/openGemini/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.OpenTable({0x14000050460?, 0x2?}, 0x103a6b9c8, 0x103a6b9d0, 0x140004808f0)
        /Users/liuhancheng/Project/openGemini/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:275 +0x130
github.com/openGemini/openGemini/engine/index/tsi.(*MergeSetIndex).Open(0x14000498680)
        /Users/liuhancheng/Project/openGemini/engine/index/tsi/mergeset_index.go:125 +0x80
github.com/openGemini/openGemini/engine/index/tsi.MergeSetOpen({0x103a3da00?, 0x14000498680?})
        /Users/liuhancheng/Project/openGemini/engine/index/tsi/mergeset_index.go:994 +0x38
github.com/openGemini/openGemini/engine/index/tsi.(*IndexRelation).IndexOpen(...)
        /Users/liuhancheng/Project/openGemini/engine/index/tsi/index_relation.go:58
github.com/openGemini/openGemini/engine/index/tsi.(*IndexBuilder).Open(0x140001b8cf0)
        /Users/liuhancheng/Project/openGemini/engine/index/tsi/index_builder.go:136 +0x9c
github.com/openGemini/openGemini/engine.(*DBPTInfo).OpenIndexes.func1({0x14000590990, 0x29})
        /Users/liuhancheng/Project/openGemini/engine/partition.go:384 +0x978
created by github.com/openGemini/openGemini/engine.(*DBPTInfo).OpenIndexes
        /Users/liuhancheng/Project/openGemini/engine/partition.go:316 +0x2c4
```

### Additional context（其他的一些补充内容）

The user can not restart the `ts-server` if this bug exists.

relative go env

```
GOVERSION="go1.20.1"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-fstack-protector-strong -D_FORTIFY_SOURCE=2 -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-O2 -g"
CGO_FFLAGS="-O2 -g"
CGO_LDFLAGS="-O2 -g"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/cl/54lb9kcx343_b_sfndckqfy00000gn/T/go-build2187912087=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

system info 

```
system_profiler SPSoftwareDataType SPHardwareDataType


Software:

    System Software Overview:

      System Version: macOS 13.2.1 (22D68)
      Kernel Version: Darwin 22.3.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Secure Virtual Memory: Enabled
      System Integrity Protection: Disabled

Hardware:

    Hardware Overview:
      Model Name: MacBook Pro
      Model Identifier: MacBookPro18,2
      Model Number: Z14X000EECH/A
      Chip: Apple M1 Max
      Total Number of Cores: 10 (8 performance and 2 efficiency)
      Memory: 64 GB
      OS Loader Version: 8419.80.7
```

**Comment by xiangyu5632** [comment]
> It seems that this bug is related to the `MacOS` operating system.
> 
> Some `mmap` function is not implemented in the darwin platform [ref](https://github.com/openGemini/openGemini/blob/02aa42be6a8d75b67f6504130c2927ae1b0ed711/lib/fileops/fs_darwin.go#L23)

fixed #146

**Comment by shilinlee** [comment]
Change this `var disableMmap = false ` to `var disableMmap = true`, because macOS is not support mmap yet!

https://github.com/openGemini/openGemini/blob/ef1999f8271c8c95603db9bd80ab383439732da5/open_src/github.com/VictoriaMetrics/VictoriaMetrics/lib/fs/reader_at.go#L10-L12

I'll fix it in next PR. We decide to disable mmap by default.

**Comment by shilinlee** [comment]
fixed by #158
